src/network/networkd-route-nexthop.c

```diff
@@ -553,30 +563,36 @@ static int append_nexthop_one(const Route *route, const RouteNextHop *nh, struct
 
         (*rta)->rta_len += sizeof(struct rtnexthop);
 
-        if (nh->family == route->family) {
-                r = rtattr_append_attribute(rta, RTA_GATEWAY, &nh->gw, FAMILY_ADDRESS_SIZE(nh->family));
-                if (r < 0)
-                        goto clear;
-
-                rtnh = (struct rtnexthop *)((uint8_t *) *rta + offset);
-                rtnh->rtnh_len += RTA_SPACE(FAMILY_ADDRESS_SIZE(nh->family));
-
-        } else if (nh->family == AF_INET6) {
-                assert(route->family == AF_INET);
-
-                r = rtattr_append_attribute(rta, RTA_VIA,
-                                            &(RouteVia) {
-                                                    .family = nh->family,
-                                                    .address = nh->gw,
-                                            }, sizeof(RouteVia));
-                if (r < 0)
-                        goto clear;
+        /* We don't have the gateway yet for device-only routes, which are AF_UNSPEC */
+        if (in_addr_is_set(nh->family, &nh->gw)) {
```

Comment:
Please drop the condition. I do not think it works.
Please adjust `nh->family` in `route_section_verify_nexthops()` if it is `AF_UNSPEC`.

Reply:
I added the family adjustment in route_section_verify_nexthops() as requested. However, I found that we still need to check in_addr_is_set() before appending RTA_GATEWAY in append_nexthop_one().

For device-only nexthops, we should not send RTA_GATEWAY at all, the kernel uses rtnh_ifindex directly without needing a gateway. If we send RTA_GATEWAY with a zero address, the kernel interprets it as "via 0.0.0.0" rather than a true device-only route, which defeats the purpose of this feature.

So the final approach is:
1. Set nh->family = route->family in verification (for hash/compare consistency after parsing)
2. Check in_addr_is_set() before RTA_GATEWAY in netlink emission (to skip gateway for device-only)